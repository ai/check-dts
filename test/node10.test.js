let { join } = require('path')

let check = require('../check')

jest.mock('worker_threads', () => {
  throw new Error('no workers')
})

async function run(fixture) {
  let stdout = { out: '' }
  stdout.write = symbols => {
    stdout.out += symbols
  }
  stdout.print = (...lines) => {
    stdout.write(lines.join('\n') + '\n')
  }
  let cwd = join(__dirname, 'fixtures', fixture)
  let exitCode = await check(stdout, cwd, stdout.print)
  return [exitCode, stdout.out]
}

async function good(fixture) {
  let [exitCode, out] = await run(fixture)
  expect(exitCode).toBe(true)
  return out
}

jest.setTimeout(20000)

it('works without web workers', async () => {
  expect(await good('simple')).toMatchSnapshot()
})
