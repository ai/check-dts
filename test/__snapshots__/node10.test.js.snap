// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works without web workers 1`] = `
"- Check types
[32mâœ”[39m Check types
[32mâœ” [39m[90mtest/[37merrors.ts[90m[39m
[32mâœ” [39m[90mtest/[37mtypes.ts[90m[39m
"
`;
